### PR TITLE
Added MySQL support for Bigint.

### PR DESCRIPTION
**Pull request by dhensby**
See #3288

https://github.com/silverstripe-labs/silverstripe-sqlite3/pull/6
https://github.com/silverstripe/silverstripe-mssql/pull/18
https://github.com/silverstripe/silverstripe-postgresql/pull/30

### CODE REVIEW

**Comment by dhensby** [comment]
Ok. I'd merge these changes if I had permission to ss-labs and also if I had any idea if the other DB drivers worked properly ;)

@tractorcow I nominate you to get this 7 month old piece of work in!

**Comment by tractorcow** [comment]
Sure,

Can you fix that indentation please?

File: model/fieldtypes/Bigint.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ */
+class Bigint extends Int {
+
+	function __construct($name, $defaultVal = 0) {
+		$this->defaultVal = is_int($defaultVal) ? $defaultVal : 0;
+
+		parent::__construct($name);
+	}
+
+	function requireField() {
```
Comment (tractorcow) [inline model/fieldtypes/Bigint.php]: `public function` would be nice too :)

Comment (dhensby) [inline model/fieldtypes/Bigint.php]: declaring method visibility?! are you sure...

**Comment by dhensby** [comment]
@tractorcow do you know why all this info is sent through with `$values` param but almost all of it is ignored and hard coded into the string for the DB column?

eg: default, null, etc....

**Comment by tractorcow** [comment]
Not really. I didn't write it, I just refactored it. :)

**Comment by dhensby** [comment]
@tractorcow  :)

**Comment by tractorcow** [comment]
I think it would be good to see a test to ensure that a mysql adaptor can store and retrieve really big numbers.

**Comment by lcobucci** [comment]
What do we need to have this merged?

**Comment by dhensby** [comment]
I need to add a test :(

**Comment by lcobucci** [comment]
And may I help you with that?

**Comment by dhensby** [comment]
@lcobucci of course. If you'd like to, just cherry-pick my commit and create a test (as requested by @tractorcow https://github.com/silverstripe/silverstripe-framework/pull/3935#issuecomment-77234985) and then open a new PR and I'll close this one

**Comment by lcobucci** [comment]
@dhensby sure :smile:

**Comment by dhensby** [comment]
@lcobucci did you ever get round to making that test?

**Comment by dhensby** [comment]
@tractorcow I'm looking at going over my old PRs that need finishing off - where do you think the best place for this test is?

File: model/fieldtypes/Bigint.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Represents a signed 8 byte integer field. Do note PHP running as 32-bit might not work with Bigint properly, as it
+ * would convert the value to a float when queried from the database since the value is a 64-bit one.
+ *
+ * @package framework
+ * @subpackage model
+ * @see Int
+ */
+class Bigint extends Int {
```
Comment (tractorcow) [inline model/fieldtypes/Bigint.php]: BigInt I think, not Bigint?

**Comment by tractorcow** [comment]
Best places for tests are in a new framework/tests/model/BigIntTest.php

**Comment by stefanhuber** [comment]
Wouldn't it be possible to use e.g. DECIMAL(20,0) as datatype for "emulating" bigint? Does anybody now if this would have any disadvantages?

**Comment by dhensby** [comment]
> Wouldn't it be possible to use e.g. DECIMAL(20,0) as datatype for "emulating" bigint? Does anybody now if this would have any disadvantages?

From very quick googling, no not much difference.

**Comment by sminnee** [comment]
@dhensby tests or close: you be the judge!

**Comment by dhensby** [comment]
@sminnee tests added at last

**Comment by tractorcow** [comment]
Will need merging up, and have bigint() added to the standard db abstract class. :)

**Pull request merged**